ui/panels-plugin/src/plugins/time-series-chart/utils/data-transform.ts

```diff
@@ -39,98 +32,14 @@ export const EMPTY_GRAPH_DATA = {
-  let timeRange: AbsoluteTimeRange | undefined = undefined;
-  const steps: number[] = [];
-  for (const { isLoading, data } of queryResults) {
-    if (isLoading || data === undefined || data.timeRange === undefined || data.stepMs === undefined) continue;
-
-    // Keep track of query steps so we can calculate a common one for the graph
-    steps.push(data.stepMs);
-
-    // If we don't have an overall time range yet, just start with this one
-    if (timeRange === undefined) {
-      timeRange = data.timeRange;
-      continue;
-    }
-
-    // Otherwise, see if this query has a start or end outside of the current
-    // time range
-    if (data.timeRange.start < timeRange.start) {
-      timeRange.start = data.timeRange.start;
-    }
-    if (data.timeRange.end > timeRange.end) {
-      timeRange.end = data.timeRange.end;
-    }
-  }
-
-  if (timeRange === undefined) return undefined;
-
-  // Use the greatest common divisor of all step values as the overall step
-  // for the x axis (or if only one query, just use that query's step value)
-  let stepMs: number;
-  if (steps.length === 1) {
-    stepMs = steps[0] as number;
-  } else {
-    const calculatedStepMs = gcd(...steps);
-    stepMs = calculatedStepMs < MIN_STEP_INTERVAL_MS ? MIN_STEP_INTERVAL_MS : calculatedStepMs;
-  }
-
-  const startMs = timeRange.start.valueOf();
-  const endMs = timeRange.end.valueOf();
-
-  return { startMs, endMs, stepMs };
-}
-
-/**
- * Given a common time scale, generates an array of timestamp (in ms) values
- * for the x axis of a graph.
- */
-export function getXValues(timeScale: TimeScale): number[] {
-  const xValues: number[] = [];
-  let timestamp = timeScale.startMs;
-  while (timestamp <= timeScale.endMs) {
-    xValues.push(timestamp);
-    timestamp += timeScale.stepMs;
-  }
-  return xValues;
-}
-
-/**
- * Given a TimeSeries from a query and the time scale, gets the values for the
- * y axis of a graph, filling in any timestamps that are missing from the time
- * series data with `null` values.
- */
-export function getYValues(series: TimeSeries, timeScale: TimeScale): Array<number | null> {
-  let timestamp = timeScale.startMs;
-
-  const yValues: Array<number | null> = [];
-  for (const valueTuple of series.values) {
-    // Fill in values up to the current series value timestamp with nulls
-    while (timestamp < valueTuple[0]) {
-      yValues.push(null);
-      timestamp += timeScale.stepMs;
-    }
-
-    // Now add the current value since timestamp should match
-    yValues.push(valueTuple[1]);
-    timestamp += timeScale.stepMs;
-  }
-
-  // Add null values at the end of the series if necessary
-  while (timestamp <= timeScale.endMs) {
-    yValues.push(null);
-    timestamp += timeScale.stepMs;
-  }
-
-  return yValues;
+export function getCommonTimeScaleForQueries(queries: RunningQueriesState): TimeScale | undefined {
+  const seriesData = queries.map((query) => (query.isLoading ? undefined : query.data));
```

Comment:
I _think_ this should have an equivalent output of how this was written before. I initially thought that just passing `query.data` was enough, but I wasn't sure if that would lead to some edge cases where old data would be used when a refetch happened, so I erred on using `undefined` when loading.

Reply:
Gotcha, yeah on my first pass this and the getCommonTimeScaleForQueries change make sense, planning to pull locally just to double check though!

Reply:
There does seem to be a glitch when you drag to zoom on a chart--the chart loads, and you briefly see the lines render, but then the loading state comes back, before rendering again

https://user-images.githubusercontent.com/9369625/229568187-38cdc631-e19f-49de-afe5-0597974bde56.mp4